### PR TITLE
JAVA-2634: Change stream documentation samples

### PR DESCRIPTION
**Pull request by jyemin**

### CODE REVIEW

File: driver/src/examples/documentation/DocumentationSamples.java [hunk 1]
```diff
@@ -603,6 +613,52 @@ public void testDeletions() {
...
     }
 
+    @Test
+    public void testWatch() {
+        assumeTrue(isDiscoverableReplicaSet() && serverVersionAtLeast(3, 5));
+
+        final MongoCollection<Document> inventory = collection;
+        final AtomicBoolean stop = new AtomicBoolean(false);
+
+        new Thread(new Runnable() {
```
Comment (jyemin) [inline driver/src/examples/documentation/DocumentationSamples.java]: Without this the test will never finish?

File: driver/src/examples/documentation/DocumentationSamples.java [hunk 2]
```diff
@@ -603,6 +613,52 @@ public void testDeletions() {
         assertEquals(0, collection.count());
     }
 
+    @Test
+    public void testWatch() {
+        assumeTrue(isDiscoverableReplicaSet() && serverVersionAtLeast(3, 5));
+
+        final MongoCollection<Document> inventory = collection;
```
Comment (jyemin) [inline driver/src/examples/documentation/DocumentationSamples.java]: In the other examples we just call it "collection" but here we call it "inventory" as in the DRIVERS ticket.  Is that what you prefer?

**Comment by jyemin** [comment]
Added @rozza 

@kay-kim any feedback?

**Comment by kay-kim** [comment]
lgtm from me as well.  There was a slack message from Rathi and Aly regarding the question posted by robert in the JIRA ticket.   And Aly was good with the examples.  See below

rathi
[14:51] 
The drivers team has some concerns about the change steam examples we are doing for the docs team : https://jira.mongodb.org/browse/DRIVERS-414
[14:51]
They feel that these examples don't represent how the feature will be used
[14:52]
Can one of you please have a look at it and come up with better examples?

Aly
[15:04] 
the examples are fine
[15:05]  
I'll be doing a deeper tutorial that covers a more in-depth view of the feature

**Pull request closed without merging**